x-pack/apm-server/sampling/processor.go

```diff
@@ -33,10 +33,11 @@ const (
 	// loggerRateLimit is the maximum frequency at which "too many groups" and
 	// "write failure" log messages are logged.
 	loggerRateLimit = time.Minute
-)
 
-// ErrStopped is returned when calling ProcessBatch on a stopped Processor.
-var ErrStopped = errors.New("processor is stopped")
+	// shutdownGracePeriod is the time that the processor has to gracefully
+	// terminate after the stop method is called.
+	shutdownGracePeriod = 500 * time.Millisecond
```

Comment:
This seems very low. If we don't wait for the full grace period every time, how would you feel about increasing this to something higher, like 5 seconds?

Reply:
Sounds good. Thanks :D